src/Shouldly/ShouldStaticClasses/ShouldCompleteInExtensions.cs

```diff
@@ -64,6 +80,25 @@ private static void CompleteIn(Task actual, TimeSpan timeout, string? customMess
             // When exception is a timeout exception we can provide a better error, otherwise rethrow
             if (inner is ShouldlyTimeoutException exception)
             {
+                // Cancel the task first if a cancellation token is provided
+                if (cancellationTokenSource != null)
+                {
+                    cancellationTokenSource.Cancel();
+                    try
+                    {
+                        actual.Wait();
+                    }
+                    catch (AggregateException aggregateException)
+                    {
+                        if (aggregateException.InnerExceptions.Count != 1 ||
+                            aggregateException.InnerException is not OperationCanceledException canceledException ||
+                            canceledException.CancellationToken != cancellationTokenSource.Token)
```

Comment:
I'm worried about this. Plenty of times, what is thrown is a TaskCanceledException with no matching CancellationToken. I just observed this with Entity Framework last week.

Reply:
I'm also not sure what I think about the blocking wait in the first place. It could legitimately deadlock in a test with a message pump like NUnit installs if you put [Apartment(STA)] on an async method.

If it's okay to not wait for the task when not using cooperative cancellation, why is it not okay to not wait for the task either way?